[Yao Yi] iP

Pulpy-y opened the pull request:
## DukeX
> “Per aspera ad astra.” – A Latin code ([source](https://en.wikipedia.org/wiki/Per_aspera_ad_astra))

DukeX frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from here.
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 
1. And it is **FREE**! :sunglasses::kiss:

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Pulpy-y committed:
Level-1

Pulpy-y committed:
Level-2 done

Pulpy-y committed:
Level-3 task done

Pulpy-y committed:
level 4 done

Pulpy-y committed:
automated testing done

Pulpy-y committed:
level 5 exception

Pulpy-y committed:
level 6 delete item

Pulpy-y committed:
added some files

Pulpy-y committed:
Create duke constructor and run method

Pulpy-y committed:
Level 7

Pulpy-y committed:
Level 8

Pulpy-y committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Pulpy-y committed:
Update git ignore

Pulpy-y committed:
Resolve merge issues

Pulpy-y committed:
Create more files and oop

Pulpy-y committed:
Package all files

Pulpy-y committed:
Add JUnit test cases

Pulpy-y committed:
Create Jar

Pulpy-y committed:
Add JavaDoc

Pulpy-y committed:
Edit according to coding standard

Pulpy-y committed:
Merge branch 'branch-A-JavaDoc'

Pulpy-y committed:
Improve on previous code

Pulpy-y committed:
Add find by keyword

Pulpy-y committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

Pulpy-y committed:
Done checkstyle

Pulpy-y committed:
Done checkstyle again

Pulpy-y committed:
Merge branch 'A-CheckStyle'

Pulpy-y committed:
Set up GUI and complete basics

Pulpy-y committed:
Remove unused codes

Pulpy-y committed:
Tidy up code and fix incomplete label bug

david-eom commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
+        db.flip();
+        return db;
+    }
```
I think the javadoc for public method should be included for these two method!

david-eom commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,46 @@
+package duke;
+import duke.command.Command;
```
I think there should be an extra blank line between `package` and `import` :))

david-eom commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,49 @@
+
+import java.util.Scanner;
+
+import duke.command.*;
```
Should avoid `import *` statements and import the relevant classes instead.

david-eom commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,60 @@
+
+    public static String WELCOME = "Hello from\n" + LOGO + "Type 'help' to see how to talk to me!";
+
+    /**
+     * Constructor of Ui.
+     */
+    public Ui() {
+    }
```
The constructor might be slightly redundant here.

david-eom commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class Command {
+
+    /**
+     * Constructor for Command.
+     */
+    public Command() {
+    }
```
Same comment as `Ui`! :))

david-eom reviewed (commented):
LGTM! The code is extremely clean and easy to read, most of my comments are just nitpicking 😂 (oops sorry for being so anal but I guess that's what we have to do for the first PR review..) Excellent code, keep up the good work!! 👍

wlren commented on .gitignore:
```diff
@@ -15,3 +16,10 @@ bin/
 
 /text-ui-test/ACTUAL.txt
 text-ui-test/EXPECTED-UNIX.TXT
+data/tasks.txt
```
Should configure build outputs to a separate folder and ignore that folder instead of cluttering the.gitignore

wlren commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,131 @@
+                    parseTask(task, tasks);
+                }
+            }
+        } catch (FileNotFoundException | DukeException e) {
```
FileNotFoundException should never be triggered according to the task requirements

wlren commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,52 @@
+            result = taskList.delete(id);
+        } catch (IndexOutOfBoundsException e) {
+            throw new OutOfBoundsException();
+        }
```
Can be put into one try block perhaps?

wlren reviewed (commented):
LGTM! Very cool implementation and clean code, could not find much errors and just pointed out some minor stuff.

Pulpy-y committed:
Add assert

Pulpy-y committed:
Merge pull request #1 from Pulpy-y/branch-A-Assertions

Add assert

Pulpy-y committed:
Improve code quality

Pulpy-y committed:
Add CI yaml file

Pulpy-y committed:
Merge pull request #2 from Pulpy-y/master

Merge master branch to Code Quality branch

Pulpy-y committed:
Merge pull request #3 from Pulpy-y/branch-A-CodeQuality

Merge code quality branch to master

Pulpy-y committed:
Merge pull request #4 from Pulpy-y/master

Merge master branch to A-CI

Pulpy-y committed:
Merge pull request #5 from Pulpy-y/branch-A-CI

Merge branch-A-CI into master

Pulpy-y committed:
Add snooze feature

The original tasks could not be postponed

Adding a snooze command that can execute with the format of taskNumber(int) + newDate(String).

Pulpy-y committed:
Improve GUI

Pulpy-y committed:
Add Ui screenshot

Pulpy-y committed:
Update README.md

Pulpy-y committed:
Create README.md

Pulpy-y committed:
Change gradle main class

Pulpy-y committed:
Merge branch 'master' of https://github.com/Pulpy-y/ip
